python/ray/tests/test_tracing.py

```diff
@@ -0,0 +1,168 @@
+import asyncio
+import glob
+import json
+import os
+import pytest
+import shutil
+import tempfile
+
+import ray
+from ray.test_utils import check_call_ray
+
+# Create temporary spans folder for trace output.
+spans_dir = tempfile.gettempdir() + "/spans"
+
+
+@pytest.fixture()
+def cleanup_dirs():
+    """Cleanup temporary spans_dir folder at beginning and end of test."""
+    if os.path.exists(spans_dir):
+        shutil.rmtree(spans_dir)
+    os.makedirs(spans_dir)
+    yield
+    # Enable tracing only sets up tracing once per driver process.
+    # We set ray.__traced__ to False here so that each
+    # test will re-set up tracing.
+    ray.__traced__ = False
+    if os.path.exists(spans_dir):
+        shutil.rmtree(spans_dir)
+
+
+@pytest.fixture()
+def ray_start_cli_tracing(scope="function"):
```

Comment:
this is not used anywhere? can we make sure to use pytest parameterize (https://docs.pytest.org/en/stable/parametrize.html) to test both cli and init workflow?

Reply:
was not able to use pytest parameterize for different fixtures, but refactored into 6 different tests.